js/calendar.js

```diff
@@ -83,3 +91,153 @@ function refreshCalendar(nextState) {
+			if (!allDay) {
+				// "dd MMM HH:MM - dd MMM HH:MM"
+				startDate += space + startHour + colon + startMinute;
+				endDate += space + startHour + colon + startMinute;
+			}
+		} else if (startDay != endDay) {
+			// same year, same month, different day - "dd - dd MMM YYYY"
+			if (allDay) {
+				startDate = startDay;
+				endDate = endDay + space + months[endMonth] + space + endYear;
+			} else {
+				// "dd MMMM YYYY HH:MM - dd MMM YYYY HH:MM"
+				startDate = startDay + space + months[startMonth] + space + startHour + colon + endMinute;
+				endDate = endDay + space + months[endMonth] + space + endHour + colon + endMinute;
+			}
+		} else {
+			// same day - "dd MMM HH:MM - HH:MM"
+			startDate = startDay + space + months[startMonth] + space + startHour + colon + startMinute;
+			endDate = endHour + colon + endMinute;
+
+		}
+		date = startDate + dash + endDate;
+
+		//	location
+		location = eventDataExtended.location;
+
+		// description 
+		description = parseDescription(eventDataExtended.description);
+
+		// Facebook event button
+		let eventFacebookLinkElement = document.getElementById("facebook-link");
+		let link = parseFacebookLink(eventDataExtended.description);
+		if (link) {
+			eventFacebookLinkElement.onclick = () => {window.open(link)};
+			eventFacebookLinkElement.style.display = "inline";
+		} else {
+			// catches null regex matches
+			eventFacebookLinkElement.style.display = "none";
+		}
+	
+
+		// set text
+		document.getElementById("event-text-title").textContent = title;
+		document.getElementById("event-text-date").textContent = date;
+		document.getElementById("event-text-location").textContent = location;
+		var descStr = "";
+		for (line in description) {
+			descStr += description[line].toString() + "<br>";
+		}
+		document.getElementById("event-text-description").innerHTML = descStr; //innerhtml since description is already string
+		document.getElementById('calendar-event').style.display = "block"; //show
+	} else {
+		//null event
+		console.log("ERROR: event is null");
+		//should I keep this?
+	}
+	
+}
+
+function hideEvent() {
+	// hide event information by hiding the element
+	document.getElementById('calendar-event').style.display = "none";
+}
+
+function parseDescription(description) {
+	var retDesc = "";
+	let regex = new RegExp(/&lt;desc&gt;(.*?)&lt;\/desc\&gt;/, "g");
+	// parse and escape for <desc></desc> tags. toString prevent XSS
+	let match = regex.exec(description.toString());
+	if (match != null) {
+		if (match.length >= 2) {
+			retDesc = match[1];
+		}
+	} else {
+		// null, so just use the description
+		retDesc = description.toString();
+	}
+	// parse and replace <br> tags with \n
+	let regexBr = new RegExp(/\<br\>/, "g");
+	return retDesc.split(regexBr);
+	//returns list of lines
+}
+
+function parseFacebookLink(description) {
+	var retLink = "";
+	let regex = new RegExp(/&lt;fb&gt;(.*?)&lt;\/fb\&gt;/);
+	// parse and escape for <fb></fb> tags. toString prevents XSS???
```

Comment:
I'm guessing that description is already a string here? Not sure, haven't had a look.

However, when this function is called, the result is used as a URL to go to and nothing else. So XSS isn't really a concern here I don't think.